src/main/java/rs/teslaris/core/service/impl/document/EventServiceImpl.java

```diff
@@ -130,6 +154,85 @@ public Page<EventIndex> searchEventsImport(List<String> names, String dateFrom,
             Pageable.ofSize(5), EventIndex.class, "events");
     }
 
+    @Override
+    public List<EventsRelationDTO> readEventRelations(Integer eventId) {
+        var event = findOne(eventId);
+
+        if (event.getSerialEvent()) {
+            throw new NotFoundException("One time event with this ID does not exist.");
+        }
+
+        var relations = new ArrayList<>(
+            eventsRelationRepository.getRelationsForOneTimeEvent(eventId).stream()
+                .map(EventsRelationConverter::toDTO).toList());
+
+        var reverseRelations =
+            eventsRelationRepository.getReverseRelationsForOneTimeEvent(eventId).stream()
+                .map(EventsRelationConverter::toDTO).toList();
+
+        relations.addAll(reverseRelations);
+
+        return relations;
+    }
+
+    @Override
+    public List<EventsRelationDTO> readSerialEventRelations(Integer serialEventId) {
+        var serialEvent = findOne(serialEventId);
+
+        if (!serialEvent.getSerialEvent()) {
+            throw new NotFoundException("Serial event with this ID does not exist.");
+        }
+
+        return eventsRelationRepository.getRelationsForSerialEvent(serialEventId).stream()
+            .map(EventsRelationConverter::toDTO).collect(Collectors.toList());
+    }
+
+    @Override
+    public void addEventsRelation(EventsRelationDTO eventsRelationDTO) {
+        if (eventsRelationDTO.getSourceId().equals(eventsRelationDTO.getTargetId())) {
+            throw new SelfRelationException("selfRelationEventError");
+        }
+
+        if (eventsRelationRepository.relationExists(eventsRelationDTO.getSourceId(),
+            eventsRelationDTO.getTargetId())) {
+            throw new ConferenceReferenceConstraintViolationException(
+                "relationAlreadyExistsError");
+        }
+
+        var sourceEvent = findOne(eventsRelationDTO.getSourceId());
+
+        if (sourceEvent.getSerialEvent()) {
+            throw new ConferenceReferenceConstraintViolationException(
+                "Source event cannot be serial.");
```

Comment:
Why here is not used a key for the message (as in the line 199 in this file)?

Reply:
Because this state is unreachable if you are accessing the functionality through our frontend, only if you are sending directly to our API.

Reply:
ok